### ISSUE TITLE
Signature parser can't resolve local functions

### ISSUE THREAD

**Issue by mateoatr** [mono/linker#1286]
There's no way to retrieve any member when passing the signature of a local function (`<ParentMethod>g__`), lambda method (`<PM>b__`), lambda display class (`<PM>c__`) or state machine type (`<PM>d__`) to the signature parser. This is somewhat problematic in the scenario in which a warning is triggered by code written inside any of these, since we are then forced to suppress warnings in the parent method.

```C#
[module: UnconditionalSuppressMessage("", "IL1234", Scope="member", Target="M:Test.Foo.{Bar}g_LocalMethod|1_0")]

namespace Test
{
	public class Foo
	{
		public void Bar
		{
			void LocalMethod()
			{
				// Code that causes ILLinker to output warning 1234.
			}

			LocalMethod();
		}
	}
}
```

**Comment by vitek-karas** [comment]
This is a C#-ism - from IL's perspective there's no such thing as local function (and so on). It would be nice for linker to be able to recognize these and "map" them to the C# syntax, but the question is how far do we take this.

**Comment by marek-safar** [comment]
I agree with Vitek it'd be better not to try to implement IL -> C# decoder as it's more complex (your list is incomplete) than it looks like and it will only get more complicated in the future.

I think the problem is with Target using C# like syntax only and we should extend that to IL syntax as UnconditionalSuppressMessage should be working with any language and not only with C#
